src/offset/mod.rs

```diff
@@ -431,6 +431,24 @@ pub trait TimeZone: Sized + Clone {
         self.timestamp_opt(secs, nanos as u32).unwrap()
     }
 
+    /// Makes a new `DateTime` from the number of non-leap microseconds
+    /// since January 1, 1970 0:00:00 UTC (aka "UNIX timestamp").
+    ///
+    /// #Example
+    /// ```
+    /// use chrono::{Utc, TimeZone};
+    ///
+    /// assert_eq!(Utc.timestamp_micros_opt(1431648000000).unwrap().timestamp(), 1431648);
+    /// ```
+    fn timestamp_micros_opt(&self, micros: i64) -> LocalResult<DateTime<Self>> {
```

Comment:
Can this also use `NaiveTime::from_timestamp_micros`?

@djc Do we want the `_opt` suffix?

Reply:
I tested using `NaiveTime::from_timestamp_micros` locally, it seems to work as expected.
Once we decide if you'll rather not have the `_opt` suffix, I'll push my changes.

Reply:
Let's skip the `opt` suffix!